app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentUnitTest.kt

```diff
@@ -0,0 +1,70 @@
+package fr.free.nrw.commons.nearby
+
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import fr.free.nrw.commons.TestCommonsApplication
+import fr.free.nrw.commons.kvstore.JsonKvStore
+import fr.free.nrw.commons.nearby.fragments.NearbyParentFragment
+import fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.MockitoAnnotations
+import org.powermock.reflect.Whitebox
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import java.lang.reflect.Method
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [21], application = TestCommonsApplication::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class NearbyParentFragmentUnitTest {
+
+    @Mock
+    private lateinit var mapView: MapView
+
+    @Mock
+    private lateinit var applicationKvStore: JsonKvStore
+
+    @Mock
+    private lateinit var mapBox: MapboxMap
+
+    @Mock
+    private lateinit var presenter: NearbyParentFragmentPresenter
+
+    private lateinit var fragment: NearbyParentFragment
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        fragment = NearbyParentFragment()
+        Whitebox.setInternalState(fragment, "mapView", mapView)
+        Whitebox.setInternalState(fragment, "applicationKvStore", applicationKvStore)
+        Whitebox.setInternalState(fragment, "mapBox", mapBox)
+        Whitebox.setInternalState(fragment, "presenter", presenter)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun `Start map without gps test when last location known`() {
```

Comment:
@Ayan-10 can you also add mock verification for all mocked objects to both of your tests, for example

```
verify(mapView).onStart()
```

you can also check for number of calls like

```
verify(mapView, times(1)).onStart()
```